tap_dynamics/discover.py

```diff
@@ -0,0 +1,79 @@
+import os
+
+import singer
+from singer import metadata
+from singer.catalog import Catalog
+
+from tap_dynamics.streams import get_streams
+
+LOGGER = singer.get_logger()
+
+
+def get_abs_path(path):
+    return os.path.join(os.path.dirname(os.path.realpath(__file__)), path)
+
+def _get_key_properties_from_meta(schema_meta):
+    return schema_meta[0].get('metadata').get('table-key-properties')
+
+def _get_replication_method_from_meta(schema_meta):
+    return schema_meta[0].get('metadata').get('forced-replication-method')
+
+def _get_replication_key_from_meta(schema_meta):
+    if _get_replication_method_from_meta(schema_meta) == 'INCREMENTAL':
+        return schema_meta[0].get('metadata').get('valid-replication-keys')[0]
+    return None
+
+def get_schemas(config, config_path):
+
+    schemas = {}
+    schemas_metadata = {}
+
+    streams = get_streams(config, config_path)
+
+    LOGGER.info('There are {:d} valid streams in MS Dynamics'.format(len(streams)))
+
+    for stream_name, stream_object in streams.items():
+        schema = stream_object.schema
+
+        meta = metadata.get_standard_metadata(
+            schema=schema,
+            key_properties=stream_object.key_properties,
+            replication_method=stream_object.replication_method
```

Comment:
You should be able to use the kwarg here for `valid_key_properties` https://github.com/singer-io/singer-python/blob/66688d736fa699beeed63af91db2fdc1f8914bb1/singer/metadata.py#L26 and you could remove the code a few lines below.